Some documentation that describes file management in DataHub including POSIX and ACL (#538)
EGI-Federation/documentation: content/en/users/data/management/datahub/file-management/_index.md

---
title: DataHub File Management

> linkTitle: DataHub-File-Management

Reviewer: ```suggestion
linkTitle: File Management
```

weight: 30
type: docs
aliases:
  - /users/datahub/datahub-file-management
description: >
  File Management in DataHub
---

## Web interface

> All files in Onedata are organized in spaces. The Web User interface allows for

Reviewer: ```suggestion
All files in DataHub are organized in spaces. The Web User interface allows for
```

uploading new files as well as opening existing files.

In order to upload a file simply open the directory in which the file should be
placed and drag the file into the browser window:

![drag the file into the browser window](datahub-drag-file.png)

Opening or downloading a file simply requires double clicking on the file in the

> file window. \*Make sure that the popups for this browser window are not

Reviewer: ```suggestion
file window. 
{{% alert title="Note" color="info" %}} Make sure that the popups for this browser window are not
```

> blocked, and unblock them if necessary.\*\*

Reviewer: ```suggestion
blocked, and unblock them if necessary.{{% /alert %}} 
```

## Direct access via POSIX

Files can also be accessed directly via POSIX protocol, using Oneclient tool.
Details on how to use are described
[here](https://onedata.org/#/home/documentation/doc/using_onedata/oneclient.html).

## CDMI (Cloud Data Management Interface)

For more advanced use cases, files can be managed using
[CDMI](http://www.snia.org/cdmi) protocol, as described in details
[here](https://onedata.org/#/home/documentation/doc/advanced/cdmi.html).

## File Permissions

You can control access to your data with a combination of:

- classical (POSIX) file permissions
- access control lists

## POSIX Permissions

> Onedata allows you to control access to your data in a POSIX-like fashion for

Reviewer: ```suggestion
DataHub allows you to control access to your data in a POSIX-like fashion for
```

users, group and others in terms of read, write and executable permissions.

An important nuance regarding file permissions is that all space members are
treated as a virtual group which is the group owner of all files in the space.
That means that whenever a file is accessed by a space member who is not the
owner of the file, the group permissions are taken into consideration.
Permissions for "others" are considered when accessing
[shares](https://onedata.org/#/home/documentation/doc/using_onedata/shares.html).

Consider the following example of file's POSIX permissions:

```shell
rwx r-- ---
 |   |   |
 |   |   guests
 |   |
 |   space members
 |
 owner user
```

In the above case, the creator of the file (its owner user) has a full access to
the file. All space members have read access to the file. Users (guests) who try
to access the file through a share will fail to do so as all permissions are
declined for "others".

In order to edit permissions:

> 1. Navigate to **Oneprovider** page

Reviewer: ```suggestion
```

1. Navigate to data management page, by click on **Data** on the left menubar

> 1. Select a **file** or a **directory** by clicking on it

Reviewer: ```suggestion
1. Select a **file** or a **directory** and right clicking on it
```

> 1. Click on the **lock icon**

Reviewer: ```suggestion
1. Click on the **Permission** option
```

1. Select **POSIX** type of permissions radio button at the top
1. Enter privileges in octal form (e.g. 770)
1. Click **OK** in order to save changes

![DataHub POSIX](datahub-posix.png)

## Access Control Lists

You can also setup permissions using more advanced Access Control Lists option
to control permissions for individual users and groups.

In order to edit ACL's:

1. Navigate to **Data** tab

> 1. Select a **file** or a **directory** by clicking on it

Reviewer: ```suggestion
1. Select a **file** or a **directory** by right clicking on it
```

> 1. Click on the **lock icon**

Reviewer: ```suggestion
1. Click on the **Permissions** option
```

---
title: DataHub File Management
linkTitle: File Management
weight: 30
type: docs
aliases:

>   - /users/datahub/datahub-file-management

Reviewer 2: What's the reasoning behind using this alias?

Author: I would say none in particular. Indeed it would correspond top the path where it will be. Do you prefer to remove it?

Reviewer 2: So the location of the file and files will define the path of the URL, so currently it will be `/users/data/management/datahub/file-management`, which looks right.
The aliases are mainly meant to use when we re-organise the repository so that we don't have broken links, and the only thing they will do will be to redirect to the real URL.
So I don't think it's appropriate for a new file.

Author: I've removed it now
